src/textual/widgets/_input.py

```diff
@@ -255,6 +255,7 @@ def cursor_width(self) -> int:
         return self._position_to_cell(len(self.value)) + 1
 
     def render(self) -> RenderableType:
+        self.view_position = self.view_position
```

Comment:
Forces validation again. This is required as the input may have changed size since last render.

Reply:
How the heck does one conclude the issue was here?

Reply:
> Forces validation again. This is required as the input may have changed size since last render.

Given that, to the casual reader, the change might look like a nop; perhaps this should be a comment in the code?

Reply:
@rodrigogiraoserrao The only difference with width:auto is that the size of the input changes after text is appended, leaving the view position in the wrong place. The `validate_view_position` should always keep the view position in the right place, so I figured i justt wasn't being called. 

@davep It's somewhat of an established pattern now. Not keen on cut and pasting the same comment everywhere.